frontend/asfui/core/settings.py

```diff
@@ -101,6 +173,200 @@
 ]
 
 
+#############################################################
+# SOCIAL AUTH IMPLEMENTATION - GOOGLE
+if SOCIAL_AUTH_GOOGLE_ENABLED:
+    SOCIALACCOUNT_STORE_TOKENS=True
+    SITE_ID = 1
+    LOGIN_REDIRECT_URL ='home'
+    SOCIALACCOUNT_QUERY_EMAIL = True
+    ACCOUNT_LOGOUT_ON_GET= True
+    ACCOUNT_UNIQUE_EMAIL = True
+    ACCOUNT_EMAIL_REQUIRED = True
+    INSTALLED_APPS += ('allauth.socialaccount.providers.google',)
+    AUTHENTICATION_BACKENDS = (
+    'django.contrib.auth.backends.ModelBackend',
+    'allauth.account.auth_backends.AuthenticationBackend',
+    )
+    SOCIALACCOUNT_PROVIDERS = {
+        'google': {
+            'SCOPE': [
+                'profile',
+                'email',
+            ],
+            'AUTH_PARAMS': {
+                'access_type': 'online',
+            }   
+        }
+    }
+
+
+#############################################################
+# SOCIAL AUTH IMPLEMENTATION - GITHUB
+
+if SOCIAL_AUTH_GITHUB_ENABLED:
+    INSTALLED_APPS +=('allauth.socialaccount.providers.github',)
+    AUTHENTICATION_BACKENDS = (
+    "django.contrib.auth.backends.ModelBackend",
+    "allauth.account.auth_backends.AuthenticationBackend",
+    )
+    
+    
+    
+#############################################################
+# SAML2 AUTH IMPLEMENTATION - Add your SSO URL and additional Configuration details below before enabling
+
+
+if SOCIAL_SAML2_ENABLED:
+    import saml2
+    import saml2.saml
+    INSTALLED_APPS += ('djangosaml2',)
+    MIDDLEWARE.append('djangosaml2.middleware.SamlSessionMiddleware')
+    CSRF_COOKIE_DOMAIN = None
+    SESSION_COOKIE_SAMESITE=None
+    SAML_SESSION_COOKIE_NAME = 'saml_session'
+    LOGIN_REDIRECT_URL = 'home'
+    SESSION_EXPIRE_AT_BROWSER_CLOSE = True
+    LOGIN_URL = '/saml2/login/'
+    URL_PREFIX = ''
+    LOGIN_EXEMPT_URLS = (r'^%ssaml2/' % URL_PREFIX,)
+    AUTHENTICATION_BACKENDS = (
+        'django.contrib.auth.backends.ModelBackend',
+        'djangosaml2.backends.Saml2Backend',
+        )
+    SAML_LOGOUT_REQUEST_PREFERRED_BINDING = saml2.BINDING_HTTP_POST
+    SAML_DJANGO_USER_MAIN_ATTRIBUTE = 'username'
+    #SAML_DJANGO_USER_MAIN_ATTRIBUTE_LOOKUP = '__iexact'
+    SAML_USE_NAME_ID_AS_USERNAME = True
+    SAML_CREATE_UNKNOWN_USER = True
+    SAML_ATTRIBUTE_MAPPING = {
+	    'NameID': ('username',),
+        'email': ('emailAddress', ),
+        'Firstname': ('Firstname', ),
+        'Lastname': ('Lastname', ),
+        }
+    
+    from os import path
+    ASF_URL = "http://127.0.0.1:8000"
+    SSO_URL = 'https://myvmware.workspaceair.com'
```

Comment:
Please avoid using any organization-specific internal domains. Please make them configurable for ASF users.

Reply:
configurable setting added